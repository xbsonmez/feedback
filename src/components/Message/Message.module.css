.message {
    display: contents;
}

.textAreaModal {
    resize: none;
    width: calc(100% - 50px);
    height: 170px;
}

.sendBtn {
    margin-top: 20px;
    height: 32px;
    width: calc(100% - 50px);
    color: white;
    background-color: var(--tyOrange);
    border: none;
}