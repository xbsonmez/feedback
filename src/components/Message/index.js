import React, { useState } from 'react';
import PropTypes from 'prop-types';
import s from './Message.module.css';

const Message = (prop) => {

  const [text, setText] = useState('');

  const { onSend } = prop;
  const onTextChange = (e) =>{
    setText(e);
  };

  const onSendText = () => {
    onSend(text);
  }

  return (
    <div className={s.message}>
         <textarea 
          data-testid="text-area"
          className={s.textAreaModal} 
          maxLength="2000"
          placeholder="Write Your Feedback"
          onChange={(e)=>{onTextChange(e.target.value)}}
         />
          <button 
          aria-label='onSend' 
          className={s.sendBtn} 
          onClick={()=>{onSendText()}}>
            GÃ¶nder
          </button>
    </div>
  );
}

Message.propTypes = {
  onSend: PropTypes.func,
};
export default Message;
