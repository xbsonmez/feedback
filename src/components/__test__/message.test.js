import React from "react";
import {render, cleanup, screen, fireEvent} from '@testing-library/react';
import Message from "../Message";
import '@testing-library/jest-dom';

describe('Message Component Test', () => {

  afterEach(cleanup);

  const initialProps = {
    onSend: jest.fn(),
   
  };
    it('Should Take A Snapshot', () => {
      const { asFragment } = render(<Message { ...initialProps }/>)
      expect(asFragment(<Message { ...initialProps }/>)).toMatchSnapshot()
    
    });
    it("render without crash", () =>{
        const div = document.createElement("div");
        render(<Message {...initialProps}/>, div);
    });

    it("onSend Click Test", () =>{
        const handleClick = jest.fn();
        render(<Message  onSend= { handleClick }/>);
        const btn = screen.getByRole('button', {
          name: /onSend/i
        });
        fireEvent.click(btn);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it("textarea rendered test", () =>{
        const div = document.createElement("div");
        render(<Message {...initialProps}/>, div);
        screen.queryByTestId(`text-area`).textContent='undefined';
        const textarea = screen.queryByTestId(`text-area`);
        expect(textarea).toHaveTextContent();
    });
});