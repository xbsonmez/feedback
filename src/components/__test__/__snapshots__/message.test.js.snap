// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component Test Should Take A Snapshot 1`] = `
<DocumentFragment>
  <div
    class="message"
  >
    <textarea
      class="textAreaModal"
      data-testid="text-area"
      maxlength="2000"
      placeholder="Write Your Feedback"
    />
    <button
      aria-label="onSend"
      class="sendBtn"
    >
      GÃ¶nder
    </button>
  </div>
</DocumentFragment>
`;
